src/librustc/middle/check_const.rs

```diff
@@ -474,9 +474,13 @@ impl<'a, 'tcx, 'v> Visitor<'v> for CheckCrateVisitor<'a, 'tcx> {
                                     self.tcx, ex, ExprTypeChecked) {
                                 Ok(_) => {}
                                 Err(msg) => {
-                                    span_err!(self.tcx.sess, msg.span, E0020,
+                                    let ret = span_err!(self.tcx.sess, msg.span, E0020,
                                               "{} in a constant expression",
-                                              msg.description())
+                                              msg.description());
+                                    span_help!(self.tcx.sess, msg.span,
+                                               "Dividing by 0 always leads to an error and is \
+                                                certainly not what you want");
```

Comment:
"certainly not what you want" doesn't sound like a nice way to phrase it. In this case, there's no single suggestion for fixing that stands out, so we shouldn't be adding this help in the first place (that's what the long diagnostics are for)